### PR TITLE
Workbox Docs

### PR DESCRIPTION
**Pull request by gauntface**
- Migrates current docs from workboxjs.org
- Adds reference doc building into gulp - requires manual running so should be safe to add (will need to test further after next workbox release).

### CODE REVIEW

File: src/content/en/tools/workbox/_toc.yaml [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 - title: Overview
   path: /web/tools/workbox/
-- title: Workbox Documentation
-  status: external
-  path: https://workboxjs.org/
-- title: Workbox Git Repository
-  status: external
+- title: Get Started
+  section:
+  - title: WebPack
```
Comment (addyosmani) [inline src/content/en/tools/workbox/_toc.yaml]: Webpack -> webpack

File: gulp-tasks/reference-docs.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
+const gulp = require('gulp');
```
Comment (addyosmani) [inline gulp-tasks/reference-docs.js]: Should this file be called workbox-reference-docs.js? or include comments near the top explaining what section this file is generating documentation for? Atm, Workbox team probably know what it does but I can see the rest of the team not.

File: src/content/en/tools/workbox/examples.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+also combined the most common pieces into one module, `workbox-sw`. This module
+supports precaching, routing of fetch events, runtime caching and more.
+
+## workbox-sw
+
+The `workbox-sw` example demonstrates a complete service worker
+implementation, demonstrating the common parts of the API. It's an in-depth
+exploration of everything you'd need to build a production-ready service worker.
+
+<a href="https://workbox-samples.glitch.me/examples/workbox-sw/" class="btn">View Example</a>
```
Comment (addyosmani) [inline src/content/en/tools/workbox/examples.md]: We should include a note (in the Readme/Contributing/elsewhere) that shares how these examples are maintained or updated. e.g are they auto-pushed to Glitch? is there a point-of-contact for getting them updated if we notice a bug etc.

File: src/content/en/tools/workbox/reference-docs/latest/_toc.yaml [hunk 4]
```diff
@@ -1,57 +1,57 @@
 toc:
 - title: "BaseCacheManager"
-  path: /web/tools/workbox/reference-docs/latest/module-workbox-precaching.BaseCacheManager
+  path: /web/tools/workbox/reference-docs/v2.0.3/module-workbox-precaching.BaseCacheManager
```
Comment (addyosmani) [inline src/content/en/tools/workbox/reference-docs/latest/_toc.yaml]: Pardon a naive question: with each generation of new reference docs, will we be preserving docs for previous releases (WF repo including each version) or removing them? (e.g 2.0.3, 2.0.4 etc)

File: README.md [hunk 5]
```diff
@@ -45,6 +45,14 @@ need to re-build those files using:
 npm run build
 ```
 
+## Start Local Server
+
+To view the site locally, just run:
+
+```
+npm run start
```
Comment (petele) [inline README.md]: You can also just do `npm start`

File: gulp-tasks/reference-docs.js (hunk 2 above)
Comment (petele) [reply gulp-tasks/reference-docs.js]: If it's going to be used for other projects later, we should keep the name, but if this is absolutely unique to workbox, then it should include something. It'd also be nice to have a small header with the appropriate description in it.

File: src/content/en/tools/workbox/community.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+---
```
Comment (petele) [inline src/content/en/tools/workbox/community.md]: Does DevSite like this yaml formatting? I didn't think it did...

File: src/content/en/tools/workbox/community.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+layout: page
+title: Community & Support
+styles:
```
Comment (petele) [inline src/content/en/tools/workbox/community.md]: Same for styles, not sure devsite supports this.

File: src/content/en/tools/workbox/community.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+---
+layout: page
```
Comment (petele) [inline src/content/en/tools/workbox/community.md]: missing `book_path:` and `project_path:`
add `page_type: reference`

File: src/content/en/tools/workbox/community.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+title: Community & Support
+styles:
+ - /styles/community.css
+---
+
+# Community & Support
```
Comment (petele) [inline src/content/en/tools/workbox/community.md]: Needs `{: .page-title }`

File: src/content/en/tools/workbox/examples.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+to let you know when two responses are different.
+
+<a href="https://workbox-samples.glitch.me/examples/workbox-broadcast-cache-update/" class="btn">View Example</a>
+
+#### workbox-cache-expiration
+
+Takes care of expiring cached entries based on the maximum number or age of
+entries.
+
+<a href="https://workbox-samples.glitch.me/examples/workbox-cache-expiration/" class="btn">View Example</a>
```
Comment (petele) [inline src/content/en/tools/workbox/examples.md]: class should be `button`

**Comment by petele** [comment]
Also, looks like something is making the gulp build process die. See https://travis-ci.org/google/WebFundamentals/builds/283279020?utm_source=github_status&utm_medium=notification

File: gulp-tasks/reference-docs.js (hunk 2 above)
Comment (gauntface) [reply gulp-tasks/reference-docs.js]: I've tried to make it generic, although the gulp command is specific to workbox at the mo `gulp reference-docs:workbox`. So basically - no not intended to be specific, but only tested on workbox.

File: src/content/en/tools/workbox/examples.md (hunk 3 above)
Comment (gauntface) [reply src/content/en/tools/workbox/examples.md]: Do you want it here or on the demos?

I think you are worrying about stuff before it's happened.

File: README.md (hunk 5 above)
Comment (gauntface) [reply README.md]: do you want me to change it to that? I just wanted it clear on the readme

File: src/content/en/tools/workbox/reference-docs/latest/_toc.yaml (hunk 4 above)
Comment (gauntface) [reply src/content/en/tools/workbox/reference-docs/latest/_toc.yaml]: At the moment there is nothing that will remove them but the build process is configured to old look for and build the latest tagged release.

File: src/content/en/tools/workbox/community.md (hunk 6 above)
Comment (gauntface) [reply src/content/en/tools/workbox/community.md]: Fixed this (I hadn't actually ported this over or linked it up)

File: README.md (hunk 5 above)
Comment (petele) [reply README.md]: It's a nit, don't care either way, i Iike npm start, it's faster.

**Comment by petele** [comment]
I added the do-not-merge tag, when this is ready to go, we'll remove it.

**Comment by gauntface** [comment]
@petele is this waiting on your review / @addyosmani review or something else?

**Pull request merged**